browser/brave_content_browser_client.cc

```diff
@@ -167,7 +168,8 @@ bool BraveContentBrowserClient::AllowAccessCookie(
   content_settings::BraveCookieSettings* cookie_settings =
       (content_settings::BraveCookieSettings*)io_data->GetCookieSettings();
   bool allow = !ShouldBlockCookie(allow_brave_shields, allow_1p_cookies,
-                   allow_3p_cookies, first_party, url) &&
+                   allow_3p_cookies, first_party, url,
+                   cookie_settings->GetAllowGoogleAuth()) &&
```

Comment:
`ShouldBlockCookie` will need a refactor but not in the scope of this issue.

Reply:
Agreed on both counts